client/proposal.go

```diff
@@ -181,9 +190,19 @@ func (c *Client) ProposeChannel(ctx context.Context, prop ChannelProposal) (*Cha
 		return nil, errors.WithMessage(err, "channel proposal")
 	}
 
-	// 3. fund
-	fundingErr := c.fundChannel(ctx, ch, prop)
-	return ch, fundingErr
+	// 3. fund the channel. If funding fails, settle the channel.
+	if err := c.fundChannel(ctx, ch, prop); err != nil {
+		c.log.WithField("channel", ch.ID()).Warn("Funding failed, will settle channel.")
+		// Increment the account usage, or we will not be able to settle since
+		// the funding failed.
+		c.wallet.IncrementUsage(ch.Params().Parts[ch.machine.Idx()])
+		cfErr := ChannelFundingError{FundingError: err}
+		if err = ch.Settle(ctx, false); err != nil {
```

Comment:
I also again thought about this after having implementing this slightly "hacky" approach with the nested errors and so on, and I think handling this unfunded error would probably be a cleaner way. 
One further advantage is that even if the settling may fail, since you are in possession of the channel object, you as the client could still call settle again.

As you said, one disadvantage is that there is rarely a use case where you want to receive an unfunded channel object, so the `ProposeChannel` method should actually take care of getting back your funds.